browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,474 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 3.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/brave_cosmetic_resources_tab_helper.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <utility>
+
+#include "base/json/json_writer.h"
+#include "base/strings/stringprintf.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_shields/browser/ad_block_custom_filters_service.h"
+#include "brave/components/brave_shields/browser/ad_block_regional_service_manager.h"
+#include "brave/components/brave_shields/browser/ad_block_service.h"
+#include "brave/components/brave_shields/browser/ad_block_service_helper.h"
+#include "brave/components/brave_shields/browser/brave_shields_util.h"
+#include "brave/content/browser/cosmetic_filters_communication_impl.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/common/chrome_isolated_world_ids.h"
+#include "content/browser/web_contents/web_contents_impl.h"
+#include "content/public/browser/global_routing_id.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/renderer/render_frame.h"
+#include "brave/components/cosmetic_filters/resources/grit/cosmetic_filters_generated_map.h"
+#include "ui/base/resource/resource_bundle.h"
+
+const char k_pre_init_script[] =
+  "(function() {"
+     "if (window.content_cosmetic == undefined) {"
```

Comment:
this will break if the page has a `content_cosmetic` global var already (e.g., `const content_cosmetic = true`).  A page could intentionally or otherwise prevent all scriptlets and cosmetic filtering.

Working with @SergeyZhukovsky now on a fix

Reply:
NVM above is incorrect, not an issue bc of isolated worlds